openmessaging-api-samples/src/main/java/io/openmessaging/examples/consumer/ConsumerExample.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 OpenMessaging
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.openmessaging.examples.consumer;
+
+import io.openmessaging.ClientDriverManager;
+import io.openmessaging.api.Client;
+import io.openmessaging.api.Message;
+import io.openmessaging.api.MessageId;
+import io.openmessaging.api.consumer.Consumer;
+import io.openmessaging.api.consumer.ConsumerConfiguration;
+import io.openmessaging.api.consumer.MessageListener;
+import io.openmessaging.api.reader.Reader;
+import io.openmessaging.api.reader.ReaderConfiguration;
+
+public class ConsumerExample {
+
+    public static void main(String[] args) throws Exception {
+        final Client client = ClientDriverManager.createClient(
+            "oms:rocketmq://localhost:10911/us-east:namespace");
+
+        ConsumerConfiguration conf = new ConsumerConfiguration();
```

Comment:
OMS.newKeyValue vs. ConsumerConfiguration

Reply:
consumer configuration extends the java properties. rather than rebuild a class KeyValue

Reply:
`KeyValue` is an ambiguous term because it literally means a **single** tuple of `(key, value)` rather than a Map, Dictionary or Properties concept.